### PR TITLE
Upgrade babel and related dependencies

### PR DESCRIPTION
**Pull request by alexandec**
## Description
Upgrade babel and related dependencies. See also the [equivalent PR](https://github.com/department-of-veterans-affairs/vets-website/pull/18325) which was merged in vets-website.

All upgrades are minors except `babel-plugin-dynamic-import-node` which is a major upgrade. This was already upgraded in vets-website without issue.

This PR removes `babel-jest` and `eslint-plugin-jest` because they are no longer used.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#28952

## Testing done
- Local build tested in Chrome and IE 11
- Successful CI and Review Instance builds

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by jhonnyoddball — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -116,12 +116,12 @@
     "axe-core": "^4.1.3",
     "babel-core": "^7.0.0-bridge.0",
     "babel-eslint": "^10.1.0",
-    "babel-jest": "^23.4.2",
-    "babel-plugin-dynamic-import-node": "^1.2.0",
+    "babel-jest": "^27.2.0",
```
Comment (Rleahy22) [inline package.json:119]: Do we need this? I don't believe we have jest installed

**Review by Rleahy22 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (alexandec) [reply package.json]: Good point! Just pushed a new commit to remove this, and eslint-plugin-jest as well

**Review by zacharymorel — APPROVED** [review]

**Comment by alexandec** [comment]
@department-of-veterans-affairs/vsa-facilities-frontend  @department-of-veterans-affairs/vsa-caregiver-frontend @department-of-veterans-affairs/vsp-testing  please review this babel upgrade PR when you get a chance

**Comment by alexandec** [comment]
@k80bowman could you please review this one as well? For some reason Jhonny's review wasn't enough to satisfy the frontend-review-group requirement

**Review by k80bowman — APPROVED** [review]

**Review by ddzz — APPROVED** [review]

**Review by mpelzsherman — APPROVED** [review]

**Review by mdewey — APPROVED** [review]
LGTM

**Pull request merged**